### PR TITLE
Audio for Dynamic memory allocation

### LINKED ISSUE #6960: Audio for Dynamic memory allocation in C
### Description

I will contribute audio in Dynamic memory allocation in C.

### Domain

C/CPP

### Type of Contribution

Audio

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by Harsh652-cpu**
<hr>

## Description 📜

In C langauge  we commonly use array and declare its size.If the computer memory is limited to very small extent then application of dynamic memory allocation in programming is must.I hope this will help you and give a boost in your coding journey.
Fixes #6960 
<hr>

## Type of change 📝

- [x] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [ ] Doc#umentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by Harsh652-cpu** [comment]
@Suvraneel I have created new issue and pr.Please check this one.

File: C_CPP/Dynamic Memory allocation/Audio_Dynamic_memory_allocation.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# DYNAMIC MEMORY ALLOCATION
+The process of allocating memory to the variables during execution of the program or at run time is known as dynamic memory allocation.Till now whenever we needed an array
+we had decleared a static array of fixed size,say
```
Comment (Suvraneel) [inline C_CPP/Dynamic Memory allocation/Audio_Dynamic_memory_allocation.md:3]:
```suggestion
we had decleared a static array of fixed size,say  
```

File: C_CPP/Dynamic Memory allocation/Audio_Dynamic_memory_allocation.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+int arr[100];
```
Comment (Suvraneel) [inline C_CPP/Dynamic Memory allocation/Audio_Dynamic_memory_allocation.md:4]:
```suggestion
```cpp
int arr[100];
```
```

File: C_CPP/Dynamic Memory allocation/Audio_Dynamic_memory_allocation.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+When this statement is executed,consecutive space for 100 integers is allocated .It is not uncommon that we may be using only 10% or 20% of the allocated space thereby wasting
+rest of the space .To overcome this problem and to utilize the memory efficiently,C language provides a mechanism of dynamically allocating memory so that only the amount
+of memory that is actually required is reserved.We reserve space only at the run time for the variables that are actually required.Dynamic memory allocation gives best
+performance in situations in which we do not know memory requirements in advance.
```
Comment (Suvraneel) [inline C_CPP/Dynamic Memory allocation/Audio_Dynamic_memory_allocation.md:8]:
```suggestion
performance in situations in which we do not know memory requirements in advance.  
```

File: C_CPP/Dynamic Memory allocation/readme.md [hunk 4]
```diff
@@ -1,2 +1,3 @@
 -[malloc,calloc,realloc](malloc,calloc,realloc.md)
```
Comment (Suvraneel) [inline C_CPP/Dynamic Memory allocation/readme.md:2]:
```suggestion
```
Remove new line

**Review by Suvraneel — CHANGES_REQUESTED** [review]
image

When you want to add a new line add 2 spaces at end of previous line, this is how md works...
If you observe the preview you'll see that none of your new lines re reflected

**Comment by Harsh652-cpu** [comment]
> image
> 
> When you want to add a new line add 2 spaces at end of previous line, this is how md works... If you observe the preview you'll see that none of your new lines re reflected

I don't want to add new line there.Please check now. @Suvraneel

**Comment by Suvraneel** [comment]
@Harsh652-cpu  Well then, you should...
Line breaks make texts more readable...
Run the text through Grammarly if needed... (Spacing needed)

**Comment by Suvraneel** [comment]
@Harsh652-cpu  Also, resolve plagiarism triggers from [here](https://www.coursehero.com/file/72519729/CS8391-notespdf/)
image
image


I hope you haven't used text from any sources in your audio as well

**Comment by Harsh652-cpu** [comment]
As it is audio contribution,its my mistake to add too many texts there.Nope I don't use direct words,I have used my own words. @Suvraneel Any more corrections?

**Comment by Suvraneel** [comment]
@Harsh652-cpu  Nope, you're mistaken... If it was video contrib you might not have to write anything...
But in audio files we suggest you mention the content roadmap at least

Example :
image

**Comment by Harsh652-cpu** [comment]
Ok sure.Check now. @Suvraneel

**Comment by Suvraneel** [comment]
LGTM

**Pull request merged**